tooling/miniutil/src/mock_write.rs

```diff
@@ -0,0 +1,76 @@
+use std::{io::Write, str::from_utf8, cell::RefCell, rc::{Rc, Weak}};
+
+use gen_minirust::libspecr::hidden::GcCompat;
+
+#[derive(Default)]
+pub struct MockBuffer {
+    buffer: Rc<RefCell<Vec<u8>>>,
+}
+
+impl MockBuffer {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub(crate) fn out(&self) -> MockWrite {
+        MockWrite { buffer: Rc::downgrade(&self.buffer) }
+    }
+
+    pub fn into_strings(self) -> Vec<String> {
+        let slice = self.buffer.borrow();
+
+        let string = from_utf8(&slice).unwrap().to_string();
+
+        string.split("\n").map( 
+            |str| str.to_string() 
+        ).collect()
+    }
+}
+
+#[derive(Default)]
+pub(crate) struct MockWrite {
```

Comment:
I am confused by why this is so complicated. A single buffer type should be enough. Just `derive(Clone)` for `MockBuffer` and then `impl Write` for it?

Also every type should come with a doc comment explaining its purpose.  The fact that this entire file has 0 comments is always a bad sign. ;)